deckhouse-controller/pkg/controller/controller.go

```diff
@@ -201,10 +265,118 @@ func (dml *DeckhouseController) runEventLoop(ec chan events.ModuleEvent) {
 				log.Errorf("Error occurred during the module %q turning off: %s", mod.GetBasicModule().GetName(), err)
 				continue
 			}
+
+		case events.ModuleStateChanged:
+			err := dml.updateModuleStatus(event.ModuleName)
+			if err != nil {
+				log.Errorf("Error occurred during the module %q status update: %s", event.ModuleName, err)
+				continue
+			}
 		}
 	}
 }
 
+func (dml *DeckhouseController) updateModuleConfigStatus(configName string) error {
+	return retry.OnError(retry.DefaultRetry, errors.IsServiceUnavailable, func() error {
+		metricGroup := fmt.Sprintf("%s_%s", "obsoleteVersion", configName)
+		dml.metricStorage.GroupedVault.ExpireGroupMetrics(metricGroup)
+		moduleConfig, err := dml.kubeClient.DeckhouseV1alpha1().ModuleConfigs().Get(dml.ctx, configName, v1.GetOptions{})
+
+		// if module config found
+		if err == nil {
+			newModuleConfigStatus := d8config.Service().StatusReporter().ForConfig(moduleConfig)
+			if (moduleConfig.Status.Message != newModuleConfigStatus.Message) || (moduleConfig.Status.Version != newModuleConfigStatus.Version) {
+				patch, err := json.Marshal([]moduleConfigPatch{{
+					Op:   "replace",
+					Path: "/status",
+					Value: v1alpha1.ModuleConfigStatus{
+						Message: newModuleConfigStatus.Message,
+						Version: newModuleConfigStatus.Version,
+					},
+				}})
+				if err != nil {
+					return err
+				}
+
+				log.Debugf(
+					"Patch /status for moduleconfig/%s: version '%s' to %s', message '%s' to '%s'",
+					moduleConfig.Name,
+					moduleConfig.Status.Version, newModuleConfigStatus.Version,
+					moduleConfig.Status.Message, newModuleConfigStatus.Message,
+				)
+
+				_, err = dml.kubeClient.DeckhouseV1alpha1().ModuleConfigs().Patch(dml.ctx, moduleConfig.Name, types.JSONPatchType, patch, v1.PatchOptions{}, "status")
```

Comment:
Why do we patch the status here? When we can change the status field in the ModuleConfig structure and then call UpdateStatus function. What's the clue?

Reply:
idk, I liked how patches work, they are immune to `resourceVersion` problem as I understand. But I see the point of not using it here.